### PR TITLE
Alexandria 05 api

### PR DESCRIPTION
**Pull request by Alexandrialexie**
__Pull Request Description__

Addresses #44 

---
__Licensing Certification__

FarmData2 is a [Free Cultural Work](https://freedomdefined.org/Definition) and all accepted contributions are licensed as described in the LICENSE.md file. This requires that the contributor holds the rights to do so. By submitting this pull request __I certify that I satisfy the terms of the [Developer Certificate of Origin](https://developercertificate.org/)__ for its contents.

### CODE REVIEW

**Comment by Alexandrialexie** [comment]
@braughtg I think this fixes homework 04.

**Comment by braughtg** [comment]
This PR does not appear to contain the API tab with solutions the the Activity 05 exercises.  Let me know if there is a different one, or be sure that you've pushed the relevant commits to this PR so that I can review them.

**Comment by Alexandrialexie** [comment]
@braughtg I had forgotten to push the branch to the origin. I have pushed them now. Thanks for letting me know!

File: farmdata2/farmdata2_modules/fd2_school/api/api.html [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+                    "SQ7",
+                ],
+
+                nextLog: '',
+                logs: [
+                   {date: "2019-02-05", area: "Chuau-1", crop: "Kale", yield: 9, units: "Bunches"},
+                   {date: "2019-03-05", area: "Chuau-1", crop: "Spinach", yield: 9, units: "Bunches"},
+                ],
+
+                farms: [
```
Comment (braughtg) [inline farmdata2/farmdata2_modules/fd2_school/api/api.html:101]: This data property does not seem to have a purpose in your code.  It should be removed.

File: farmdata2/farmdata2_modules/fd2_school/api/api.html [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        var harvReport = new Vue({
+            el: '#harvest-report',
+            data: {
+                idToCropMap: new Map(),
+                idToAreaMap: new Map(),
+                repTitle: 'My Sample Harvest Report',
+                startDate: "2020-05-05",
+                endDate: "2020-05-15",
+                crop: "Kale",
+                crops: [
```
Comment (braughtg) [inline farmdata2/farmdata2_modules/fd2_school/api/api.html:83]: This array has been replaced by the computed property, so it is no longer used.  It should be removed from your code.

File: farmdata2/farmdata2_modules/fd2_school/api/api.html [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                    "Kale",
+                    "Broccoli",
+                    "Peas",
+                    "Spinach",
+                ],
+                fields: [
```
Comment (braughtg) [inline farmdata2/farmdata2_modules/fd2_school/api/api.html:89]: This array has been replaced by the computed property, so it is no longer used. It should be removed from your code.

File: farmdata2/farmdata2_modules/fd2_school/api/api.html [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+                        this.idToAreaMap = theMap;
+                        console.log("area Map worked");
+                    })
+                    .catch((err) => {
+                        // An error occurred during the request, process it here.
+                        console.log("Error with area map");
+                        console.log(error);
+                    })
+                },
+
```
Comment (braughtg) [inline farmdata2/farmdata2_modules/fd2_school/api/api.html:180]: Seems like some excessive blank lines here.  Try to use blank lines like you would use paragraph breaks in normal writing - to separate logically distinct sections of the code.

**Comment by braughtg** [comment]
HW05 - FD2 School 05 - Web APIs

* Overall:
  - [x] - 05 Worksheet is sufficiently complete
  - [x] - Code in `.module` file is neat
  - [ ] - Code in `api.html` file is neat
  - [ ] - Extraneous `console.log` statements have been cleaned up.
  - [ ] - Unused code (due to changes) has been cleaned up.
  - [ ] - Made PR with descriptive body. (11)
  - [x] - PR Body contains link to the issue. (11)
  - [x] - Commits made for nameable units of work. 
  - [x] - Commit messages are descriptive.
  - [ ] - PR Converted from draft to regular (42)

* API Sub-tab contents:
  - [x] - Add API sub-tab (9)
  - [x] - Farm, User, Language are bound to Vue Data (12,14)
  - [x] - “Generate Report” `@click` uses Axios API call to get farm info (18)
  - [x] - `@click` sets Farm, User, Language in Vue Data (22)
  - [x] - `created()` hook puts cropToIDMap into VueData (29,30)
  - [x] - computed property returns array of crop names (36,37)
  - [x] - crop dropdown populated from computed property (38)

* Comments:
  * See additional comment in the code on your PR.
    * More consistent indentation, spacing and alignment will make your code easier to read.
    * Unused code that is left over from earlier implementations that is no longer needed should be cleaned up.
    * Console.log statements used for incremental development or debugging should be removed once the program is working.
  * Your score will be posed in Moodle.

**Pull request closed without merging**